docs/proposals/PROPOSAL_E.md

```diff
@@ -184,6 +189,36 @@ For registries that do not support the `referrers` API, a tag MUST be pushed for
 - Periodic garbage collection may be performed by clients pushing new referrers, deleting stale referrers that have been replaced with newer versions, and tags that no longer point to an accessible manifest.
 - Clients can verify the registry does not support the `referrers` API by querying the API and checking for a 404.
 
+#### Client Expectations
+
+##### Creating Artifacts
+
+- For portability, clients should generate artifacts using image-spec until all registries where the artifact is pushed to have been upgraded.
```

Comment:
This is like saying all image tools should use the docker manifest until all registries support OCI manifests. 
What would be interesting is capturing the behavior for round tripping between a registry that does/doesn't support the new manifest type.
Line 200 captures what I think you're targeting.

Reply:
Line 200 handles what to do if the new API is not available. But if you push an unknown manifest to the registry, it will reject it from any API. If the client knows how to round trip the artifact, then why not push the artifact as an image-spec and skip the conversion and digest change on the content? Do that round tripping process need to be standardized so an artifact changed by one tool is readable by another? I was trying to avoid getting into that complexity because it feels like a time sink that would distract us from getting this finished.